tensornetwork/ncon_interface_test.py

```diff
@@ -38,14 +57,38 @@ def test_order_spec(backend):
   result = ncon_interface.ncon([a, a], [(-1, 1), (1, -2)],
                                con_order=[1],
                                backend=backend)
+
   np.testing.assert_allclose(result, np.ones((2, 2)) * 2)
 
   result = ncon_interface.ncon([a, a], [(-1, 1), (1, -2)],
                                con_order=[1],
-                               out_order=[-1, -2])
+                               out_order=[-1, -2],
+                               backend=backend)
+
   np.testing.assert_allclose(result, np.ones((2, 2)) * 2)
 
 
+def test_node_order_spec(backend):
+  node = tn.Node(np.ones((2, 2)), backend=backend)
+  result = ncon_interface.ncon([node, node], [(-1, 1), (1, -2)],
```

Comment:
Passing two of the exact same node seems like a really bad test case. 

Also, what should `ncon` do if two nodes are connect? Right now it ignores them and `ncon` just uses the underlying tensor, but that feels sketchy too.

Reply:
I think that's what it should do.  a call to `ncon` has to specify the whole contraction or throw an exception otherwise. This should override any preexisting connections on nodes. In essence, it's like using `Node.tensor` except that you don't have to access `Node.tensor` yourself.